### ISSUE TITLE
Calling Serial1.print before Serial1.begin causes white breathing led?

### ISSUE THREAD

**Issue by dmiddlecamp** [spark/firmware#228]
I vaguely recall this being fixed for "Serial", but maybe it wasn't also fixed in "Serial1" ?

Thread here:
https://community.spark.io/t/call-of-serial-print-before-serial-begin-gives-a-white-breathing-led/5070

Thanks!
David

**Comment by dmiddlecamp** [comment]
Adding to this issue from a community post:

```
#pragma SPARK_NO_PREPROCESSOR
#include "application.h"

class A {

    public:

    A()  {  Serial.begin(9600); Serial.print("die");  }

};


A anA;

void setup()
{
    // Doing Serial.print right here before Serial.begin does not kill the core.
    //Serial.begin(9600);
}

void loop() { }
```

**Comment by m-mcgowan** [comment]
Reproduced this today. 

This is a classic symptom of C++ constructors called in arbitrary order (across modules.)

So when A is initialized, the main `Serial` instance has not been initialized. The problem here is that `write` is a virtual function,yet the vtable has yet to be initialized (done in the constructor perhaps?) so that the write() method doesn't behave as expected.

Technically it is possible to fix. Each global class instance should be a static variable declared inside an accessor function. 

```
extern USBSerial& _serial();
#define Serial _serial()
```

And in the spark_wiring_usbserial.cpp

```
USBSeria& _serial() {
   static USBSerial& instance;
   return instance;
}
```

It's a bit hacky since it introduces a macro into the namespace. However, it does work. Tested and verified.

However, I'm uncertain if we should fix this bug - it's far simpler to just move the Serial.begin()/print() to the setup() method where it's usually done. The fix could potentially negatively affect more users than the bug itself.

**Comment by technobly** [comment]
This was implemented in v0.6.0 so use of Serial.begin() in a constructor on the Photon/P1/Electron no longer fails.  https://github.com/spark/firmware/blob/develop/wiring/src/spark_wiring_usbserial.cpp#L141-L146
